appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/interceptors/TimeoutInterceptor.java

```diff
@@ -190,15 +189,15 @@ private Object timeout(InvocationContext invocationContext) throws Exception {
             logger.log(Level.INFO, "No config could be found", ex);
         }
         
-        long value = (Long) FaultToleranceCdiUtils.getOverrideValue(
+        long value = FaultToleranceCdiUtils.getOverrideValue(
                 config, Timeout.class, "value", invocationContext, Long.class)
                 .orElse(timeout.value());
         // Look for a String and cast to ChronoUnit - Use the Common Sense Convertor
-        ChronoUnit unit = (ChronoUnit) FaultToleranceCdiUtils.getOverrideValue(
-                config, Timeout.class, "unit", invocationContext, String.class)
+        ChronoUnit unit = FaultToleranceCdiUtils.getOverrideValue(
+                config, Timeout.class, "unit", invocationContext, String.class).map(ChronoUnit::valueOf)
```

Comment:
Note again the `map` to `ChronoUnit` assuming `config` returns a `String`.

Reply:
See previous comment